BaboViolent2/Code/FileIO.h

```diff
@@ -60,10 +60,10 @@ class FileIO
 	virtual unsigned char * getUByteArray(int size);
 	virtual int getInt(); // short
 	virtual unsigned int getUInt(); // unsigned short
-	virtual long getLong();
-	virtual long * getLongArray(int size);
-	virtual unsigned long getULong();
-	virtual unsigned long * getULongArray(int size);
```

Comment:
Yes this code works when server reads the maps. I tested it. The general problem in that long is defined in C++ spec as "the same size as int or bigger". And it turns out that on x86 platform it have size 4 but on x64 it is 8 bytes long. And this breaks all disk and network IO. Using types from stdint header guarantees you that size always still the same.